src/main/scala/com/couchbase/spark/sql/DefaultSource.scala

```diff
@@ -74,16 +76,15 @@ class DefaultSource
       case SaveMode.Overwrite => StoreMode.UPSERT
     }
 
-    data
+    val datas = data
       .toJSON
       .map(rawJson => {
         val encoded = JsonObject.fromJson(rawJson)
         val id = encoded.get(idFieldName).toString
         encoded.removeKey(idFieldName)
         JsonDocument.create(id, encoded)
       })
-      .saveToCouchbase(bucketName, storeMode)
-
+    new DocumentRDDFunctions[JsonDocument](datas.toJavaRDD).saveToCouchbase(bucketName, storeMode)
```

Comment:
Here also I tried to resolve it by looking what the other method needs and how can I provide that !
What should we do here Please let me know what changes are required !